Impact of Multiple Measured Elements to Conditional Triples (#276)
ietf-rats-wg/draft-ietf-rats-corim: draft-ietf-rats-corim.md

{::include cddl/stateful-environment-record.cddl}
~~~

The series is an array of `conditional-series-record` that has a `selection` and an `addition`, both expressed as a list of `measurement-map`.

> The `selection` and `addition` operate within the scope of the `conditional-endorsement-series-triple-record`'s matching `condition`.

Reviewer: ```suggestion
The `selection` and `addition` operate within the scope of the environment-map found within the `conditional-endorsement-series-triple-record`'s `condition`.
```

For each `conditional-series-record` entry, if the `selection` matches the ACS entry, the `addition` is added to the ACS.

The first `conditional-series-record` entry that successfully matches the ACS entry terminates the series.
For a `conditional-series-record` to match, every measurement in the `measurement-map` list MUST match a measurement in the ACS entry.

If none of the `selection` values match in the ACS entry, the triple is not matched, and no `addition` values are accepted.

> If the `authorized-by` value is present in the triple `condition` (i.e., in the `measurement-map` of the `stateful-environment-record`), all authority entries of the `condition` MUST be present in the ACS entry, otherwise the ACS entry does not match.

Reviewer: Is this needed here>
Should we be discussing how to transform external encoding into ECT encoding so that the text in 8.9 (which already says this) can be used?

Reviewer 2: > Is this needed here> Should we be discussing how to transform external encoding into ECT encoding so that the text in 8.9 (which already says this) can be used?

I think it's OK to describe the semantics of the triple in terms of what to expect will be reflected in the ACS. The transformation section describes how to map from external to internal. The Phase 4 processing section describes how the ACS realizes an ending state. This section describes what to expect in terms of the ending state in ACS - i.e., it's a what vs. how difference.

{::include cddl/stateful-environment-record.cddl}
~~~

The series is an array of `conditional-series-record` that has a `selection`, and an `addition`, both expressed as a list of `measurement-map`.

> The `selection` and `addition` operate within the scope of the conditional endorsement series triple record's matching `condition`.

Author: I have made it more explicit and crisp, which is I suppose you wanted. Please check!

> For each `conditional-series-record` entry, if the `selection` matches the chosen ACS entry, the `addition` is added to the ACS.

Reviewer 2: > Addressed: Remove the chosen term!

The original text was trying to capture the detail that the selection scope is within the range of the `condition` and that the `addition` is within the same range.  This is a necessity since the series records don't have a qualifying environment-map.

~~~

The series object is an array of `conditional-series-record` that has both Reference and Endorsed Values.

> Each `conditional-series-record` has measurements belonging to one or more measured elements, pertaining to the environment that appears in the stateful-environment record.

Reviewer 2: ```suggestion
For each `conditional-series-record` entry, if the `selection` matches the chosen ACS entry, the `addition` is added to the ACS.
```

> The Endorsed Values are accepted if the series condition in a `conditional-series-record` matches the ACS.

Reviewer 2: ```suggestion

```

Author: Why have you deleted this?

Author: ok, understood the concept that in series, the first condition match terminates the series, to line 1224 is not really applicable!

> The first `conditional-series-record` that successfully matches an ACS Entry terminates the matching and the corresponding Endorsed Values are accepted. In order for a `conditional-series-record` to match, measurements pertaining to every measured element in the record MUST match.

Reviewer 2: ```suggestion
The first `conditional-series-record` entry that successfully matches the chosen ACS entry terminates the series. 
For a `conditional-series-record` to match, every measurement in the `measurement-map` list MUST match a measurement in the chosen ACS entry.
```

> If none of the series conditions match an ACS Entry, the triple is not matched,

Reviewer 2: ```suggestion
If none of the `selection` values match in the chosen ACS entry, the triple is not matched,
```

Reviewer 3: Are we expressing the semantics in terms of the triple or in terms of the relation of internal representations we translate triples to? I think we should be consistent and use the internal representation for defining semantics.

> and no Endorsed values are accepted.

Reviewer 2: ```suggestion
and no `addition` values are accepted.
```

> The `authorized-by` value in `measurement-map` in the stateful environment, if present,

Reviewer 2: ```suggestion
If the `authorized-by` value is present in the triple `condition` (i.e., in the `measurement-map` of the `stateful-environment-record`),
all authority entries MUST be present in the ACS entry, otherwise the ACS entry does not match.
```

> applies to reference value measurements in the triple, for `conditional-series-record` records.

Reviewer 2: ```suggestion
If the series `selection` populates `authorized-by`, the chosen ACS MUST contain the same measurements and authority as contained in the `selection` entry.
```

Reviewer 3: I think we want to say matching authority so we can punt on the matching semantics. Sounds like you want subset ([⊆](https://www.compart.com/en/unicode/U+2286)) and Andy wants nonempty intersection (often notated with [⋈](https://www.compart.com/en/unicode/U+22C8)). This is a good place to use tagging for pattern operators so both forms can be expressible.

> If the `authorized-by` value is provided in the `measurement-map` for endorsed values, they get added to the ACS.

Reviewer 2: ```suggestion
If the series `addition` entry contains `authorized-by` values, they are ignored.
```

Reviewer 2: Note: authority semantics are still being discussed. It is safer for this triple to ignore them in the addition until the semantics for series authority is determined.

Reviewer 3: I think since it's a non-extensible type, you can use a type like element-map (but a copy to not confuse internal and external representations) to drop the authorities. You can add them back with a spec update.

Author: ok, for now I will accept this suggestion , but really feel this is required. We will discuss the same in a separate PR
